apps/ngrx-1/src/app/store/status/status.selectors.ts

```diff
@@ -1,17 +1,39 @@
-import { createFeatureSelector, createSelector } from '@ngrx/store';
+import { createSelector } from '@ngrx/store';
 import { ActivityType } from '../activity/activity.model';
-import { statusFeatureKey, StatusState } from './status.reducer';
-
-export const selectStatusState =
-  createFeatureSelector<StatusState>(statusFeatureKey);
+import { selectActivities } from '../activity/activity.selectors';
+import { selectUser } from '../user/user.selectors';
+import { Status } from './status.model';
 
 export const selectStatuses = createSelector(
-  selectStatusState,
-  (state) => state.statuses
+  selectUser,
+  selectActivities,
+  (user, activities): Status[] => {
+    if (user?.isAdmin) {
+      return activities.reduce((status: Status[], activity): Status[] => {
+        const index = status.findIndex((s) => s.name === activity.type);
+        if (index === -1) {
+          return [
+            ...status,
+            { name: activity.type, teachers: [activity.teacher] },
+          ];
+        } else {
+          status[index].teachers.push(activity.teacher);
+          return status;
+        }
+      }, []);
+    }
+    return [];
+  }
 );
 
+export const selectTeachersMap = createSelector(selectStatuses, (statuses) => {
+  const map = new Map();
+  statuses.forEach((s) => map.set(s.name, s.teachers));
+  return map;
+});
+
 export const selectAllTeachersByActivityType = (name: ActivityType) =>
   createSelector(
-    selectStatusState,
-    (state) => state.teachersMap.get(name) ?? []
+    selectTeachersMap,
+    (teachersMap) => teachersMap.get(name) ?? []
```

Comment:
you don't need the last two selectors. You should create a viewModel with construct your final object that you will use in your view